node/tests/SharedTests.ts

```diff
@@ -1233,6 +1233,184 @@ export function runBaseTests<Context>(config: {
+                const key2 = "{key}-2" + uuidv4();
+                const initialCursor = "0";
+                const defaultCount = 20;
+                const resultCursorIndex = 0;
+                const resultCollectionIndex = 1;
+
+                // Setup test data - use a large number of entries to force an iterative cursor.
+                const numberMap: Record<string, string> = {};
+
+                for (let i = 0; i < 50000; i++) {
+                    numberMap[i.toString()] = "num" + i;
+                }
+
+                const charMembers = ["a", "b", "c", "d", "e"];
+                const charMap: Record<string, string> = {};
+
+                for (let i = 0; i < charMembers.length; i++) {
+                    charMap[charMembers[i]] = i.toString();
+                }
+
+                // Empty set
+                let result = await client.hscan(key1, initialCursor);
+                expect(result[resultCursorIndex]).toEqual(initialCursor);
+                expect(result[resultCollectionIndex]).toEqual([]);
+
+                // Negative cursor
+                result = await client.hscan(key1, "-1");
+                expect(result[resultCursorIndex]).toEqual(initialCursor);
+                expect(result[resultCollectionIndex]).toEqual([]);
+
+                // Result contains the whole set
+                expect(await client.hset(key1, charMap)).toEqual(
+                    charMembers.length,
+                );
+                result = await client.hscan(key1, initialCursor);
+                expect(result[resultCursorIndex]).toEqual(initialCursor);
+                expect(result[resultCollectionIndex].length).toEqual(
+                    Object.keys(charMap).length * 2, // Length includes the score which is twice the map size
+                );
+
+                const resultArray = result[resultCollectionIndex];
+                const resultKeys = [];
+                const resultValues: string[] = [];
+
+                for (let i = 0; i < resultArray.length; i += 2) {
+                    resultKeys.push(resultArray[i]);
+                    resultValues.push(resultArray[i + 1]);
+                }
+
+                // Check if all keys from charMap are in resultKeys
+                const allKeysIncluded = resultKeys.every(
+                    (key) => key in charMap,
+                );
+                expect(allKeysIncluded).toEqual(true);
+
+                const allValuesIncluded = Object.values(charMap).every(
+                    (value) => value in resultValues,
+                );
+                expect(allValuesIncluded).toEqual(true);
+
+                result = await client.hscan(key1, initialCursor, {
+                    match: "a",
+                });
+
+                expect(result[resultCursorIndex]).toEqual(initialCursor);
+                expect(result[resultCollectionIndex]).toEqual(["a", "0"]);
+
+                // Result contains a subset of the key
```

Comment:
This comment isn't really helpful.
Instead, describe what this next test is doing.  
Here you are setting up testing data by adding a large dataset to the hash at key1.